File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        s = set(nums)
        for i, n1 in enumerate(nums):
            diff = target - n1
            if diff not in s:
                continue
            if n1 == diff and nums.count(n1) < 2:
                continue
            j = nums[i+1:].index(diff) + i + 1
            return [i, j]

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for (int i = 0; i < nums.size(); i++)
        {
            int n1 = nums[i];
            for (int j = i+1; j < nums.size(); j++)
            {
                int n2 = nums[j];
                if (n1 + n2 == target)
                {
                    vector<int> ans{i, j};
                    return ans;
                }
            }
        }
        return {};
    }
};